app/src/main/java/com/arygm/quickfix/ui/search/Announcement.kt

```diff
@@ -104,6 +106,69 @@ fun AnnouncementScreen(
     }
   }
 
+  val sheetState = rememberModalBottomSheetState()
+
+  // Function to reset announcement parameters
+  val resetAnnouncementParameters = {
+    title = ""
+    category = ""
+    description = ""
+    location = null
+    titleIsEmpty = true
+    descriptionIsEmpty = true
+    // categoryIsSelected = false
+    locationIsSelected = false
+    navigationActions.saveToCurBackStack("selectedLocation", null)
+  }
+
+  // Function to update user profile
+  val updateUserProfileWithAnnouncement: (Announcement) -> Unit = { announcement ->
+    profileViewModel.fetchUserProfile(userId) { profile ->
+      if (profile is UserProfile) {
+        val announcementList = profile.announcements + announcement.announcementId
+
+        profileViewModel.updateProfile(
+            UserProfile(profile.locations, announcementList, profile.wallet, profile.uid),
+            onSuccess = {
+              accountViewModel.fetchUserAccount(profile.uid) { account ->
+                loggedInAccountViewModel.setLoggedInAccount(account!!)
+              }
+            },
+            onFailure = { e ->
+              Log.e("ProfileViewModel", "Failed to update profile: ${e.message}")
+            })
+      } else {
+        Log.e("Wrong profile", "Should be a user profile")
+      }
+    }
+  }
+
+  // Function to handle successful image upload
+  val handleSuccessfulImageUpload: (String, List<String>) -> Unit =
```

Comment:
After reviewing all your code and understanding the logic, I think you did an excellent job modularizing it.